main/network.c

```diff
@@ -859,54 +859,48 @@ php_socket_t php_network_connect_socket_to_host(const char *host, unsigned short
 		if (sa) {
 			/* make a connection attempt */
 
-			if (bindto) {
-				struct sockaddr *local_address = NULL;
-				int local_address_len = 0;
-
+			while (bindto) {
```

Comment:
While `while (bindto)` looks confusing, because it's not supposed to loop and just used for the `break`. I'd personally prefer keeping the `goto` or at least use `if (bindto) do { } while (0);`, which is the usual pattern we use for "break-only loops". Of course, it would be best if it's possible to have this in a separate function, so that early return can be used.